hazelcast-jet-spring/src/main/java/com/hazelcast/jet/spring/JetSpringServices.java

```diff
@@ -0,0 +1,297 @@
+    ) {
+        return e -> e.filterUsingService(beanServiceFactory(beanName), predicate::test);
+    }
+
+    /**
+     * Returns a transform function for the {@link StreamStage#apply(FunctionEx)}
+     * which adds a filter stage using the specified bean.
+     *
+     * @param beanName     the name of the bean
+     * @param requiredType the class of the bean
+     * @param predicate    the predicate
+     * @param <T>          the type of the bean
+     */
+    public static <I, T> FunctionEx<StreamStage<I>, StreamStage<I>> filterStreamUsingSpringBean(
+            @Nonnull String beanName,
+            @Nonnull Class<T> requiredType,
+            @Nonnull BiPredicateEx<? super T, ? super I> predicate
+    ) {
+        return e -> e.filterUsingService(beanServiceFactory(beanName, requiredType), predicate::test);
+    }
+
+    /**
+     * Returns a transform function for the {@link StreamStage#apply(FunctionEx)}
+     * which adds a filter stage using the specified bean.
+     *
+     * @param requiredType the class of the bean
+     * @param predicate    the predicate
+     * @param <T>          the type of the bean
+     */
+    public static <I, T> FunctionEx<StreamStage<I>, StreamStage<I>> filterStreamUsingSpringBean(
+            @Nonnull Class<T> requiredType,
+            @Nonnull BiPredicateEx<? super T, ? super I> predicate
+    ) {
+        return e -> e.filterUsingService(beanServiceFactory(requiredType), predicate::test);
+    }
+
+    /**
+     * Returns a transform function for the {@link StreamStage#apply(FunctionEx)}
+     * which adds a filter stage using the specified bean.
+     *
+     * @param beanName  the name of the bean
+     * @param predicate the predicate
+     * @param <T>       the type of the bean
+     */
+    public static <I, T> FunctionEx<StreamStage<I>, StreamStage<I>> filterStreamUsingSpringBean(
+            @Nonnull String beanName,
+            @Nonnull BiPredicateEx<? super T, ? super I> predicate
+    ) {
+        return e -> e.filterUsingService(beanServiceFactory(beanName), predicate::test);
+    }
+
+    /**
+     * Returns a factory that provides the specified bean as the service object.
+     *
+     * @param beanName     the name of the bean
+     * @param requiredType the class of the bean
+     * @param <T>          the type of the bean
+     */
+    public static <T> ServiceFactory<?, T> beanServiceFactory(@Nonnull String beanName, @Nonnull Class<T> requiredType) {
+        return ServiceFactory.withCreateContextFn(ctx -> new BeanExtractor())
+                             .withCreateServiceFn((c, g) -> g.getBean(beanName, requiredType));
+    }
+
+    /**
+     * Returns a factory that provides the specified bean as the service object.
+     *
+     * @param requiredType the class of the bean
+     * @param <T>          the type of the bean
+     */
+    public static <T> ServiceFactory<?, T> beanServiceFactory(@Nonnull Class<T> requiredType) {
```

Comment:
javadoc is too basic. needs some sample and more detailed documentation on how this works